include/kalmar_runtime.h

```diff
@@ -922,6 +923,10 @@ struct rw_info
         /// 2. Because the data pointer cannout be released, erase itself from devs
         if (HostPtr)
             synchronize(false);
+        if (curr) {
+            // Wait issues a system-scope release:
+            curr->wait();
```

Comment:
Yes.  The case was that an array_view destructor might deallocate the memory while it was still dirty in the L2 cache.  When L2 was later released, data was written to stale physical addresses and GPUVM faults could occur.  this check assures we flush L2 before deallocating the memory.  

At some point we should refactor the array_view copy routines so they share common code with the accelerator_view::copy routines appropriately.